app/models/user.rb

```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
   has_many :likes, foreign_key: :author_id
   has_many :comments, foreign_key: :author_id
   has_many :posts, foreign_key: :author_id
```

Comment:
- [ ] When I try to sign up, I get this error:

![image](https://github.com/otmanTR/BlogAppRoR/assets/77106962/8b3c6262-9854-4e47-8101-dfac7ea98972)

Due to the validation you set for the `posts_counter` attribute, we will get this error any time we sign up. I would advise you to find a way to assign posts_counter a number every time you are about to create a User instance. For example, you can do this in the app/models/user.rb 👍 😄:

```ruby
before_validation :assign_posts_counter

  def assign_posts_counter
    self.posts_counter = posts.count
  end
```